<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Data package for Expression Weighted Celltype Enrichment *EWCE* • ewceData</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Data package for Expression Weighted Celltype Enrichment *EWCE*">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ewceData</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.17.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/ewceData.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/neurogenomics/ewceData/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Data package for Expression Weighted Celltype
Enrichment <em>EWCE</em>
</h1>
                        <h4 data-toc-skip class="author">Alan Murphy,
Hiranyamaya (Hiru) Dash and Nathan Skene</h4>
            
            <h4 data-toc-skip class="date">2025-10-28</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/neurogenomics/ewceData/blob/master/vignettes/ewceData.Rmd" class="external-link"><code>vignettes/ewceData.Rmd</code></a></small>
      <div class="hidden name"><code>ewceData.Rmd</code></div>

    </div>

    
    
<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  document.querySelector("h1").className = "title";
});
</script><script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  var links = document.links;  
  for (var i = 0, linksLength = links.length; i < linksLength; i++)
    if (links[i].hostname != window.location.hostname)
      links[i].target = '_blank';
});
</script><div class="section level2">
<h2 id="citation">Citation<a class="anchor" aria-label="anchor" href="#citation"></a>
</h2>
<p>If you use the EWCE package, please cite</p>
<p><a href="https://www.frontiersin.org/articles/10.3389/fnins.2016.00016/full" class="external-link">Skene,
et al. Identification of Vulnerable Cell Types in Major Brain Disorders
Using Single Cell Transcriptomes and Expression Weighted Cell Type
Enrichment.Front. Neurosci, 2016.</a></p>
<p>If you use the cortex/hippocampus single cell data associated with
this package then please cite the following papers:</p>
<p><a href="http://www.sciencemag.org/content/early/2015/02/18/science.aaa1934.abstract" class="external-link">Zeisel,
et al. Cell types in the mouse cortex and hippocampus revealed by
single-cell RNA-seq. Science, 2015.</a></p>
</div>
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>The <em>EWCE</em> package is designed to facilitate expression
weighted celltype enrichment analysis as described in our Frontiers in
Neuroscience paper<span class="citation"><sup>1</sup></span>.</p>
<p>The package was originally designed to work with the single cell
cortical transcriptome data from the Linnarsson lab<span class="citation"><sup>2</sup></span> which is available at <a href="http://linnarssonlab.org/cortex/" class="external-link uri">http://linnarssonlab.org/cortex/</a>. Using this package it
is possible to read in any single cell transcriptome data, provided that
you have a cell by gene expression matrix (with each cell as a seperate
column) and a seperate annotation dataframe, with a row for each
cell.</p>
<p>The <em>EWCE</em> process involves testing for whether the genes in a
target list have higher levels of expression in a given cell type than
can reasonably be expected by chance. The probability distribution for
this is estimated by randomly generating gene lists of equal length from
a set of background genes.</p>
<p>The <em>EWCE</em> method can be applied to any gene list. In the
paper we reported it’s application to genetic and transcriptomic
datasets, and in this vignette we detail how this can be done.</p>
</div>
<div class="section level2">
<h2 id="data">Data<a class="anchor" aria-label="anchor" href="#data"></a>
</h2>
<p>The data in the <code>ewceData</code> package are used throughout
EWCE’s associated vignette to give insight into its functionality. These
datasets are:</p>
<ul>
<li><p><strong>cortex_mrna</strong> - Karolinska Cortex/Hippocamus
dataset</p></li>
<li><p><strong>hypothalamus_mrna</strong> - Karolinska Hypothalamus
dataset</p></li>
<li><p><strong>celltype_data</strong>/<strong>ctd</strong> - Subset of
the genes from Karolinska cortex/hippocampus and hypothalamus single
cell transcriptome datasets.</p></li>
<li><p><strong>example_genelist</strong> - A list of genes genetically
associated with Alzheimer’s disease.</p></li>
<li><p><strong>mouse_to_human_homologs</strong> - Table of
Human–&gt;Mouse orthologs for all human genes, containing the MGI and
HGNC symbols, Human and Mouse Entrez and Ensembl gene IDs for all human
orthologs for mouse genes. Whenin the mouse genes are defined based on a
list of all MGI markers from the <a href="http://www.informatics.jax.org/" class="external-link">MGI website</a>.</p></li>
<li>
<p>Example tables of differential expression. A list of genes found
to be differentially expressed in:</p>
<ul>
<li><p><strong>tt_alzh_BA36</strong> - the BA36 in Alzheimer’s
disease.</p></li>
<li><p><strong>tt_alzh_BA44</strong> - the BA44 in Alzheimer’s
disease.</p></li>
<li><p><strong>tt_alzh</strong> - the BA46 in Alzheimer’s
disease.</p></li>
</ul>
</li>
</ul>
<p>All the data can be loaded by using the dataset name:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/neurogenomics/ewceData" class="external-link">ewceData</a></span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ensembl_transcript_lengths_GCcontent</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ensembl_transcript_lengths_GCcontent.html">ensembl_transcript_lengths_GCcontent</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">mouse_to_human_homologs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/mouse_to_human_homologs.html">mouse_to_human_homologs</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">all_mgi_wtEnsembl</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/all_mgi_wtEnsembl.html">all_mgi_wtEnsembl</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">all_mgi</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/all_mgi.html">all_mgi</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">all_hgnc_wtEnsembl</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/all_hgnc_wtEnsembl.html">all_hgnc_wtEnsembl</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">all_hgnc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/all_hgnc.html">all_hgnc</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">example_genelist</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/example_genelist.html">example_genelist</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">tt_alz</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/tt_alzh.html">tt_alzh</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">tt_alzh_BA36</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/tt_alzh_BA36.html">tt_alzh_BA36</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">tt_alzh_BA44</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/tt_alzh_BA44.html">tt_alzh_BA44</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">ctd</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ctd.html">ctd</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">schiz_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/schiz_genes.html">schiz_genes</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">hpsd_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/hpsd_genes.html">hpsd_genes</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">rbfox_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/rbfox_genes.html">rbfox_genes</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">id_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/id_genes.html">id_genes</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">cortex_mrna</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cortex_mrna.html">cortex_mrna</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">hypothalamus_mrna</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/hypothalamus_mrna.html">hypothalamus_mrna</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">alzh_gwas_top100</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/alzh_gwas_top100.html">alzh_gwas_top100</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">mgi_synonym_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/mgi_synonym_data.html">mgi_synonym_data</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>This package also contains a single cell transcriptome (SCT) data
from <em>“Cell types in the mouse cortex and hippocampus revealed by
single-cell RNA-seq”, Science, 2015</em> with the expression and
annotation data, formatted as a CTD object.</p>
<p>Important note: The code (available in inst/scripts/make-data.R) to
derive this data can be used on other downloaded paper’s data. Note you
do NOT have to format your input single cell data like the Linnarsson
data. Just read it into R such that you have an expression matrix and an
annotation data frame. The three columns that you must have in the
annotation data frame are “cell_id”, “level1class” and
“level2class”.</p>
<p>To check the data, we can quickly plot the distribution of expression
of a given gene across all the cell types:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://wilkelab.org/cowplot/" class="external-link">cowplot</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/get_theme.html" class="external-link">theme_set</a></span><span class="op">(</span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/theme_cowplot.html" class="external-link">theme_cowplot</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cortex_mrna_dt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cortex_mrna.html">cortex_mrna</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">gene</span><span class="op">=</span><span class="st">"Necab1"</span></span>
<span><span class="va">cellExpDist</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>e<span class="op">=</span><span class="va">cortex_mrna_dt</span><span class="op">$</span><span class="va">exp</span><span class="op">[</span><span class="va">gene</span>,<span class="op">]</span>,</span>
<span>                          l1<span class="op">=</span><span class="va">cortex_mrna_dt</span><span class="op">$</span><span class="va">annot</span><span class="op">[</span></span>
<span>                            <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">cortex_mrna_dt</span><span class="op">$</span><span class="va">exp</span><span class="op">)</span>,<span class="op">]</span><span class="op">$</span><span class="va">level1class</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">cellExpDist</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html" class="external-link">geom_boxplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">l1</span>,y<span class="op">=</span><span class="va">e</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">"Cell type"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"Unique Molecule Count"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">90</span>, hjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span> </span></code></pre></div>
<p><img src="ewceData_files/figure-html/unnamed-chunk-2-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body" entry-spacing="0" line-spacing="2">
<div id="ref-skene_2016" class="csl-entry">
<div class="csl-left-margin">1. </div>
<div class="csl-right-inline">Skene, N. &amp; Grant, S. Identification of
vulnerable cell types in major brain disorders using single cell
transcriptomes and expression weighted cell type enrichment.
<em>Frontiers in Neuroscience</em> (2016). doi:<a href="https://doi.org/10.3389/fnins.2016.00016" class="external-link">10.3389/fnins.2016.00016</a>
</div>
</div>
<div id="ref-zeisel2015cell" class="csl-entry">
<div class="csl-left-margin">2. </div>
<div class="csl-right-inline">
<span class="nocase">Zeisel, A. <em>et
al.</em></span> Cell types in the mouse cortex and hippocampus revealed
by single-cell RNA-seq. <em>Science</em> <strong>347,</strong> 1138–1142
(2015).</div>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Alan Murphy, Hiranyamaya Dash, Nathan Skene.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

      </footer>
</div>






  </body>
</html>
